cardano-testnet/src/Testnet/Start/Cardano.hs

```diff
@@ -122,14 +146,38 @@ startTimeOffsetSeconds = if OS.isWin32 then 90 else 15
 -- > │   └── node-spo{1,2,3}
 -- > └── utxo-keys
 -- >     └── utxo{1,2,3}.{addr,skey,vkey}
-cardanoTestnet :: CardanoTestnetOptions -> Conf -> H.Integration TestnetRuntime
-cardanoTestnet testnetOptions Conf {tempAbsPath} = do
-  testnetMinimumConfigurationRequirements testnetOptions
-  void $ H.note OS.os
-  currentTime <- H.noteShowIO DTC.getCurrentTime
-  let tempAbsPath' = unTmpAbsPath tempAbsPath
+cardanoTestnet :: ()
+  => CardanoTestnetOptions -- ^ The options to use. Must be consistent with the genesis files.
+  -> Conf
+  -> UTCTime -- ^ The starting time. Must be the same as the one in the shelley genesis.
+  -> ShelleyGenesis StandardCrypto -- ^ The shelley genesis to use, for example 'Defaults.defaultShelleyGenesis'.
+                                   --   Some fields are overridden by the accompanying 'CardanoTestnetOptions'.
+  -> AlonzoGenesis -- ^ The alonzo genesis to use, for example 'Defaults.defaultAlonzoGenesis'.
+  -> ConwayGenesis StandardCrypto -- ^ The conway genesis to use, for example 'Defaults.defaultConwayGenesis'.
```

Comment:
Here we mix `CardanoTestnetOptions` and genesis files. So callers must be careful to provide coherent data. If they don't, they will get caught by the new checks below. Ideally, in the future, we'll want to remove fields from `CardanoTestnetOptions`, so that the intersection with the data of genesis files becomes empty. This a bigger change though, so I chose this version, which is an improvement already, and minimizes disturbing users.